contrib/internal/httputil/trace.go

```diff
@@ -53,8 +54,12 @@ func TraceAndServe(h http.Handler, cfg *TraceConfig) {
 	span, ctx := tracer.StartSpanFromContext(cfg.Request.Context(), "http.request", opts...)
 	defer span.Finish(cfg.FinishOpts...)
 
-	cfg.ResponseWriter = wrapResponseWriter(cfg.ResponseWriter, span)
-	httpinstr.WrapHandler(h, span).ServeHTTP(cfg.ResponseWriter, cfg.Request.WithContext(ctx))
+	// Wrap the handler with the appsec instrumentation when enabled
+	if appsec.Enabled() {
+		h = httpsec.WrapHandler(h, span)
+	}
+
```

Comment:
And here.

Reply:
Fixed in https://github.com/DataDog/dd-trace-go/pull/1042/commits/89a0faab579118328372ae6761d324707ccbad3c